include/effects/matrix/PatternSMBlurringColors.h

```diff
@@ -189,68 +203,50 @@ class PatternSMBlurringColors : public LEDStripEffect
     {
         g()->Clear();
 
-        enlargedObjectNUM = (Scale + 5U); // / 99.0 * (AVAILABLE_BOID_COUNT ) ;
+        enlargedObjectNUM = (Scale + 5U); // / 99.0 * (AVAILABLE_BOID_COUNT) ;
         if (enlargedObjectNUM > AVAILABLE_BOID_COUNT)
             enlargedObjectNUM = AVAILABLE_BOID_COUNT;
 
-        // deltaValue = 10; // количество зарождающихся частиц за 1 цикл
-        deltaValue =
-            trackingOBJECT_MAX_COUNT / (sqrt(CENTER_X_MAJOR * CENTER_X_MAJOR + CENTER_Y_MAJOR * CENTER_Y_MAJOR) * 4U) +
-            1U; // 4 - это потому что за 1 цикл частица пролетает ровно четверть
-                // расстояния между 2мя соседними пикселями
+        deltaValue = powder_item_max_count /
+            (sqrt(CENTER_X_MAJOR * CENTER_X_MAJOR + CENTER_Y_MAJOR * CENTER_Y_MAJOR) * 4U) + 1U;
+            // 4 - this is because in 1 cycle the particle flies exactly a
+            // quarter the distance between 2 neighboring pixels
 
-        for (int i = 0; i < trackingOBJECT_MAX_COUNT; i++)
-        {
-            trackingObjectPosX[i] = 0;
-            trackingObjectPosY[i] = 0;
-            trackingObjectSpeedX[i] = 0;
-            trackingObjectSpeedY[i] = 0;
-            trackingObjectHue[i] = 0;
-            trackingObjectState[i] = 0;
-            trackingObjectIsShift[i] = false;
-        }
```

Comment:
This is exactly the logic that needs to stay in - the array of `PowderItem` objects needs to be completely cleared _on every invocation of `Start()`_ to prevent the "dot in the corner"; it's also one of the main things I reintroduced to Supernova in #512. Which is why I moved the "init to zero/false" to a `Clear()` function in `DebrisItem` and invoke that on every instance in Supernova's `Start()`

Reply:
I just pushed an implementation of this.